crates/fuel-gas-price-algorithm/src/v1/tests/algorithm_v1_tests.rs

```diff
@@ -28,3 +32,114 @@ fn calculate__returns_sum_of_da_and_exec_gas_prices() {
     let expected = starting_exec_gas_price + starting_da_gas_price;
     assert_eq!(expected, actual);
 }
+
+fn _worst_case__correctly_calculates_value(
+    new_exec_price: u64,
+    new_da_gas_price: u64,
+    for_height: u32,
+    block_horizon: u32,
+    exec_price_percentage: u64,
+    da_gas_price_percentage: u64,
+) {
+    // given
+    let algorithm = AlgorithmV1 {
+        new_exec_price,
+        exec_price_percentage,
+        new_da_gas_price,
+        da_gas_price_percentage,
+        for_height,
+    };
+
+    // when
+    let target_height = for_height.saturating_add(block_horizon);
+    let actual = algorithm.worst_case(target_height);
+
+    // then
+    let mut expected_exec_price = new_exec_price;
+    let mut expected_da_gas_price = new_da_gas_price;
+
+    for _ in 0..block_horizon {
+        let change_amount = expected_exec_price
+            .saturating_mul(exec_price_percentage)
+            .saturating_div(100);
+        expected_exec_price = expected_exec_price.saturating_add(change_amount);
+
+        let change_amount = expected_da_gas_price
+            .saturating_mul(da_gas_price_percentage)
+            .saturating_div(100);
+        expected_da_gas_price = expected_da_gas_price.saturating_add(change_amount);
+    }
+
+    let expected = expected_exec_price.saturating_add(expected_da_gas_price);
+
+    dbg!(actual, expected);
+    assert!(actual >= expected);
+}
+
+proptest! {
+    #[test]
+    fn worst_case__correctly_calculates_value(
+        exec_price: u64,
+        da_price: u64,
+        starting_height: u32,
+        block_horizon in 0..10_000u32,
+        exec_percentage: u64,
+        da_percentage: u64,
+    ) {
+        _worst_case__correctly_calculates_value(exec_price, da_price, starting_height, block_horizon, exec_percentage, da_percentage);
+    }
+}
+
+proptest! {
+    #[test]
+    fn worst_case__never_overflows(
+        exec_price: u64,
+        da_price: u64,
+        starting_height: u32,
+        block_horizon in 0..10_000u32,
+        exec_percentage: u64,
+        da_percentage: u64,
+    ) {
+        // given
+        let algorithm = AlgorithmV1 {
+            new_exec_price: exec_price,
+            exec_price_percentage: exec_percentage,
+            new_da_gas_price: da_price,
+            da_gas_price_percentage: da_percentage,
+            for_height: starting_height,
+        };
+
+        // when
+        let target_height = starting_height.saturating_add(block_horizon);
+        let _ = algorithm.worst_case(target_height);
+
+        // then
+        // doesn't panic with an overflow
+    }
+}
+
+#[test]
+fn worst_case__same_block_gives_new_exec_price() {
```

Comment:
This name is wrong, it's misssing the da portion in the name. I think instead just saying it gives the same as calculate() would be best, for both the name and how it calculates the expected value.

Reply:
Names updated in https://github.com/FuelLabs/fuel-core/pull/2219/commits/c55aa6569f1a001544c96562dd4fd16561b9be26